referrer-policy/generic/referrer-policy-test-case.js

```diff
@@ -0,0 +1,144 @@
+var ReferrerPolicyTestCase = function(scenario, testDescription) {
+  // Check if scenario is valid.
+  test(function() {
+    // We extend the exsiting test_expansion_schema not to kill performance by
+    // copying.
+    var expectedFields = SPEC_JSON["test_expansion_schema"];
+    expectedFields["referrer_policy"] = SPEC_JSON["referrer_policy_schema"];
+
+    for (var field in expectedFields) {
+      assert_own_property(scenario, field,
+                          "The scenario contains field " + field)
+      assert_in_array(scenario[field], expectedFields[field],
+                      "Scenario's " + field + " is one of: " +
+                      expectedFields[field].join(", ")) + "."
+    }
+
+    // Check if the protocol is matched.
+    assert_equals(scenario["source_protocol"] + ":", location.protocol,
+                  "Protocol of the test page should match the scenario.")
+
+  }, "[ReferrerPolicyTestCase] The test scenario is valid.");
```

Comment:
I don't think it's worth doing this on every run; it makes sense to verify that we've generated sane code once, but it doesn't make sense to pay that cost in processing time every time we run any test.

I'd suggest either removing it right now, or adding a `TODO(burnik)` to kill it once you're confident that the generated code is correct.